custom_components/tuya_local/devices/moes_motionsensor_switch.yaml

```diff
@@ -0,0 +1,127 @@
+# "102": 0}
+# Additional DPS 44 "Illuminance State" and 47 "Motion State". I couldn't
+# figure out what those did, device debugger appears to indicate that these
+# were set exactly once (to "on") when the switch was powered on and does NOT
+# seem to trigger or respond when Illuminance or Motion trigger is met, but
+# maybe I just don't know how to check that well. A final DPS shows up only in
+# the Tuya Cloud API: 103 "Timer". There are "timers" in the Smart Life App for
+# this device but I couldn't figure out how they work.
+# Illuminance Switch: Enables/disables illuminance threshold for motion
+#                     activation
+# Motion Switch: Enables/disables motion-triggered switch
+# Motion Auto-Off Delay: Time in seconds after which to turn off light if
+#                        triggered by motion
+# Appointment Time: Time of day in minutes when motion switch stops activating,
+#                   0 for always
+# Target Time: Time of day in minutes when motion switch can activate, 0 for
+#              always
+name: MOES Motion Sensor Smart Switch
+products:
+  - id: vd5jgg8vgdbaqerq
+primary_entity:
+  entity: switch
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+secondary_entities:
+  - entity: select
+    category: config
+    name: Restore power state
+    dps:
+      - id: 14
+        name: option
+        type: string
+        mapping:
+          - dps_val: memory
+            value: Remember Last Status
+          - dps_val: "on"
+            value: "On"
+          - dps_val: "off"
+            value: "Off"
+  - entity: select
+    name: Indicator Light
+    category: config
+    dps:
+      - id: 15
+        name: option
+        type: string
+        mapping:
+          - dps_val: none
+            value: Indicator LED off
+          - dps_val: relay
+            value: Indicate switch on/off
+  - entity: switch
+    name: Illuminance Switch
+    dps:
+      - id: 43
+        name: switch
+        type: boolean
+  - entity: select
+    name: Illuminance Threshold
+    category: config
+    dps:
+      - id: 45
+        name: option
+        type: string
+        mapping:
+          - dps_val: default
+            value: Default Threshold
+          - dps_val: now
+            value: Current Illuminance
+  - entity: switch
+    name: Motion Switch
+    dps:
+      - id: 46
+        name: switch
+        type: boolean
+  - entity: select
+    name: Motion Sensitivity
+    category: config
+    dps:
+      - id: 48
+        name: option
+        type: string
+        mapping:
+          - dps_val: "low"
+            value: "Low"
+          - dps_val: "middle"
+            value: "Medium"
+          - dps_val: "high"
+            value: "High"
+  - entity: number
+    name: Motion Auto-Off Delay
+    category: config
+    dps:
+      - id: 49
+        name: value
+        type: integer
+        unit: s
+        range:
+          min: 5
+          max: 3600
+  - entity: number
+    name: Appointment Time
```

Comment:
More obvious naming is needed for these two times.  I would never guess that "Appointment" is the end of allowed hours, and "Target" is the beginning.  Maybe "Sensing start time" and "Sensing end time" would be better.
I would also add the following to make the display in hours rather than minutes, so for example 7pm shows as "19.0" rather than "1140".  7:30 will show as "19.5", which isn't ideal, but I think still better than "1170".
```
          mapping:
            - scale: 60
```